### PR TITLE
Gateway tasks and overview

### PR DESCRIPTION
**Pull request by freddygv**
This PR adds:
- A tasks page that shows at a high level what gateways are useful for different jobs to be done
- A gateway overview page, which mirrors that content but has the gateway names as headers

### CODE REVIEW

File: website/pages/docs/connect/gateways/index.mdx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+For more information ingress gateways see the [complete documentation](/docs/connect/gateways/ingress-gateway)
+and the [ingress gateway guide](https://learn.hashicorp.com/consul/developer-mesh/ingress-gateways).
+
+![Ingress Gateway Architecture](/img/ingress-gateways.png)
+
+## Terminating Gateways
+-> **1.8.0+:** This feature is available in Consul versions 1.8.0 and newer.
+
+Terminating gateways enable connections from services in the Consul service mesh to services that do not have
+sidecar proxies. These may be services running on legacy infrastructure or managed cloud services running on
```
Comment (dnephin) [inline website/pages/docs/connect/gateways/index.mdx:51]: > services that do not have sidecar proxies

Do "connect native services" have sidecars? Is this different from ingress and mesh gateways, or could this say "outside the Consul service mesh" ?

File: website/pages/docs/connect/tasks.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+description: >-
+  Tasks related to connectivity into, out of, and between Consul service meshes.
+---
+
+# Tasks
+
+~> **Note**: The features shown below are extensions of Consul’s service mesh capabilities. If you are not utilizing
+Consul service mesh then these features will not be relevant to your task.
+
+## Service-to-service traffic between service datacenters
```
Comment (dnephin) [inline website/pages/docs/connect/tasks.mdx:14]: "between Consul datacenters" ?

Comment (freddygv) [inline website/pages/docs/connect/tasks.mdx:14]: Yep, good catch.

```suggestion
## Service-to-service traffic between Consul datacenters
```

File: website/pages/docs/connect/gateways/index.mdx (hunk 1 above)
Comment (freddygv) [reply website/pages/docs/connect/gateways/index.mdx]: That's a good point. Connect native services do not have sidecars. 

Updated the wording to account for that.

File: website/pages/docs/connect/gateways/index.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+For more information ingress gateways see the [complete documentation](/docs/connect/gateways/ingress-gateway)
+and the [ingress gateway guide](https://learn.hashicorp.com/consul/developer-mesh/ingress-gateways).
+
+![Ingress Gateway Architecture](/img/ingress-gateways.png)
+
+## Terminating Gateways
+-> **1.8.0+:** This feature is available in Consul versions 1.8.0 and newer.
+
+Terminating gateways enable connections from services in the Consul service mesh to services outside the mesh.
```
Comment (blake) [inline website/pages/docs/connect/gateways/index.mdx:50]:
```suggestion
Terminating gateways enable connectivity from services in the Consul service mesh to services outside the mesh.
```

File: website/pages/docs/connect/tasks.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+For more information ingress gateways see the [complete documentation](/docs/connect/gateways/ingress-gateway)
+and the [ingress gateway guide](https://learn.hashicorp.com/consul/developer-mesh/ingress-gateways).
+
+![Ingress Gateway Architecture](/img/ingress-gateways.png)
+
+## Traffic from services in the Consul service mesh to external services
+-> **1.8.0+:** This feature is available in Consul versions 1.8.0 and newer.
+
+Terminating gateways enable connections from services in the Consul service mesh to services outside the mesh.
```
Comment (blake) [inline website/pages/docs/connect/tasks.mdx:53]:
```suggestion
Terminating gateways enable connectivity from services in the Consul service mesh to services outside the mesh.
```

File: website/pages/docs/connect/gateways/index.mdx [hunk 5]
```diff
@@ -0,0 +1,65 @@
+---
+layout: docs
+page_title: Gateways
+sidebar_title: Gateways
+description: >-
+  Gateways enable connectivity into, out of, and between Consul service meshes.
+---
+
+# Gateways
```
[the remark this answers is not in the document]
Comment (freddygv) [reply website/pages/docs/connect/gateways/index.mdx:9]: Added a brief intro, let me know what you think.

File: website/pages/docs/connect/connectivity-tasks.mdx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+isn't feasible. One scenario where this is useful is when connecting networks with overlapping IP address space.
+
+These gateways operate by sniffing the SNI header out of the mTLS connection and then routing the connection to the
+appropriate destination based on the server name requested. The data within the mTLS session is not decrypted by
+the Gateway.
+
+As of Consul 1.8.0, mesh gateways can also forward gossip and RPC traffic between Consul servers. 
+This is enabled by [WAN federation via mesh gateways](/docs/connect/gateways/wan-federation-via-mesh-gateways).
+
+For more information mesh gateways see the [complete documentation](/docs/connect/gateways/mesh-gateway)
```
Comment (blake) [inline website/pages/docs/connect/connectivity-tasks.mdx:28]:
```suggestion
For more information about mesh gateways see the [complete documentation](/docs/connect/gateways/mesh-gateway)
```

File: website/pages/docs/connect/connectivity-tasks.mdx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+Ingress gateways are an entrypoint for outside traffic. They enable potentially unauthenticated ingress traffic from
+services outside the Consul service mesh to services inside the service mesh. 
+
+These gateways allow you to define what services should be exposed, on what port, and by what hostname. You configure
+an ingress gateway by defining a set of listeners that can map to different sets of backing services.
+
+Ingress gateways are tightly integrated with Consul’s L7 configuration and enable dynamic routing of HTTP requests by
+attributes like the request path.
+
+For more information ingress gateways see the [complete documentation](/docs/connect/gateways/ingress-gateway)
```
Comment (blake) [inline website/pages/docs/connect/connectivity-tasks.mdx:45]:
```suggestion
For more information about ingress gateways see the [complete documentation](/docs/connect/gateways/ingress-gateway)
```

File: website/pages/docs/connect/connectivity-tasks.mdx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+infrastructure you do not control.
+
+Terminating gateways effectively act as egress proxies that can represent one or more services. They terminate Connect
+mTLS connections, enforce Consul intentions, and forward requests to the appropriate destination.
+
+These gateways also simplify authorization from dynamic service addresses. Consul’s intentions determine whether
+connections through the gateway are authorized. Then traditional tools like firewalls or IAM roles can authorize the
+connections from the known gateway nodes to the destination services.
+
+For more information terminating gateways see the [complete documentation](/docs/connect/gateways/terminating-gateway)
```
Comment (blake) [inline website/pages/docs/connect/connectivity-tasks.mdx:65]:
```suggestion
For more information about terminating gateways see the [complete documentation](/docs/connect/gateways/terminating-gateway)
```

File: website/pages/docs/connect/gateways/index.mdx (hunk 5 above)
[the remark this answers is not in the document]
Comment (blake) [reply website/pages/docs/connect/gateways/index.mdx:9]: LGTM!

File: website/pages/docs/connect/gateways/index.mdx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+isn't feasible. One scenario where this is useful is when connecting networks with overlapping IP address space.
+
+These gateways operate by sniffing the SNI header out of the mTLS connection and then routing the connection to the
+appropriate destination based on the server name requested. The data within the mTLS session is not decrypted by
+the Gateway.
+
+As of Consul 1.8.0, mesh gateways can also forward gossip and RPC traffic between Consul servers. 
+This is enabled by [WAN federation via mesh gateways](/docs/connect/gateways/wan-federation-via-mesh-gateways).
+
+For more information mesh gateways see the [complete documentation](/docs/connect/gateways/mesh-gateway)
```
Comment (blake) [inline website/pages/docs/connect/gateways/index.mdx:31]:
```suggestion
For more information about mesh gateways see the [complete documentation](/docs/connect/gateways/mesh-gateway)
```

File: website/pages/docs/connect/gateways/index.mdx [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+Ingress gateways are an entrypoint for outside traffic. They enable potentially unauthenticated ingress traffic from
+services outside the Consul service mesh to services inside the service mesh. 
+
+These gateways allow you to define what services should be exposed, on what port, and by what hostname. You configure
+an ingress gateway by defining a set of listeners that can map to different sets of backing services.
+
+Ingress gateways are tightly integrated with Consul’s L7 configuration and enable dynamic routing of HTTP requests by
+attributes like the request path.
+
+For more information ingress gateways see the [complete documentation](/docs/connect/gateways/ingress-gateway)
```
Comment (blake) [inline website/pages/docs/connect/gateways/index.mdx:48]:
```suggestion
For more information about ingress gateways see the [complete documentation](/docs/connect/gateways/ingress-gateway)
```

File: website/pages/docs/connect/gateways/index.mdx [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+infrastructure you do not control.
+
+Terminating gateways effectively act as egress proxies that can represent one or more services. They terminate Connect
+mTLS connections, enforce Consul intentions, and forward requests to the appropriate destination.
+
+These gateways also simplify authorization from dynamic service addresses. Consul’s intentions determine whether
+connections through the gateway are authorized. Then traditional tools like firewalls or IAM roles can authorize the
+connections from the known gateway nodes to the destination services.
+
+For more information terminating gateways see the [complete documentation](/docs/connect/gateways/terminating-gateway)
```
Comment (blake) [inline website/pages/docs/connect/gateways/index.mdx:68]:
```suggestion
For more information about terminating gateways see the [complete documentation](/docs/connect/gateways/terminating-gateway)
```

**Pull request merged**